### PR TITLE
Word limit on user biography

### PR DESCRIPTION
**Pull request by Ana06**
Issue #965

### CODE REVIEW

**Comment by Ana06** [comment]
Sorry that I closed and reopen the pull request. I had read it wrong at the beginning and I checked that the biography hasn't got more than 150 characters instead of 150 words.

File: spec/models/user_spec.rb [hunk 1]
```diff
@@ -66,6 +66,31 @@
...
+          lobortis metus id, tristique scelerisque nisl. Suspendisse potenti.
+          Suspendisse ac metus magna. Integer lobortis pharetra eros euismod
+          fringilla. Phasellus vitae orci vel magna laoreet mattis non eu neque.
+          Mauris ac dictum leo. Nullam dapibus convallis molestie. Integer
+          dignissim massa at odio feugiat tempus. Pellentesque ultrices rutrum
+          eros, a pellentesque lorem auctor in. Suspendisse sollicitudin dolor
+          vitae justo dignissim, a condimentum turpis molestie. Aenean
+          scelerisque, arcu eu congue mollis, nibh nulla finibus.
+        EOS
+        expect{create(:user, biography: long_text)}.to_not change { User.count }
```
Comment (sonalkr132) [inline spec/models/user_spec.rb]: I think we can make it more apparant that it is a validation test if it was something like:
```Ruby
expect(build(:user, biography: long_text)).to_not be_valid
```
And please put it along with other [validation tests](https://github.com/Ana06/osem/blob/biography/spec/models/user_spec.rb#L14). It is a method indeed, but I think it is a validation method.

**Comment by Ana06** [comment]
Done :wink:

**Comment by differentreality** [comment]
We had removed that validation in a previous PR, not sure why.
https://github.com/openSUSE/osem/pull/963/files#diff-4676c008b11a5480d73d4a6de01e45b9L202

**Comment by sonalkr132** [comment]
> We had removed that validation in a previous PR, not sure why.

Cause the method just existed there and it was not used (for validation).

File: app/models/user.rb [hunk 2]
```diff
@@ -168,4 +170,13 @@ def setup_role
       self.is_admin = true
     end
   end
+
+  ##
+  # Check if biography has an allowed number of words. Used as validation.
+  #
+  def check_biography_length
+    if self.biography.present?
+      errors.add(:biography, 'is limited to 150 words.') unless self.biography.split.length <= 150
```
Comment (differentreality) [inline app/models/user.rb]: if > 150  should also do.

**Comment by differentreality** [comment]
@sonalkr132 right, I see what happened. At some point we fixed the js that wasn't working, but apparently the validation wasn't in place already for some reason.
@Ana06 I would introduce the same function name back (biography_limit).

**Comment by Ana06** [comment]
I have changed the function name and I have solved the problem @differentreality pointed out in #965.

**Comment by hennevogel** [comment]
LGTM! Thanks Ana

**Pull request merged**